### PR TITLE
feat: add nav_link method to ApplicationHelper

### PR DESCRIPTION
**Pull request by ezhil56x**
### Changes
- Added the nav_link method to ApplicationHelper.
- Updated the nav_link method to include hover and selected states.

/claim #428

### CODE REVIEW

**Comment by ezhil56x** [comment]
@Shpigford Can you review this PR once?

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -50,4 +50,16 @@ def format_currency(number, options = {})
 
     number_to_currency(number, options)
   end
+
+  def nav_link(text, path, icon, current_page)
```
Comment (robzolkos) [inline app/helpers/application_helper.rb:54]: A more specific name for this particular nav link may be better in-case there are other nav related links?  Perhaps `main_nav_link` ?

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -50,4 +50,16 @@ def format_currency(number, options = {})
...
+    classes = 'block border border-transparent rounded-xl p-2 text-sm font-medium text-gray-500 flex items-center'
+    classes += ' hover:bg-white hover:border-[#141414]/[0.07] hover:text-gray-900 hover:shadow-xs' unless current_page?(path)
+  
+    content_tag(:li) do
+      link_to(path, class: classes) do
```
Comment (robzolkos) [inline app/helpers/application_helper.rb:59]:
```suggestion
    classes = 'block border border-transparent rounded-xl p-2 text-sm font-medium text-gray-500 flex items-center'
    selected_classes = 'hover:bg-white hover:border-[#141414]/[0.07] hover:text-gray-900 hover:shadow-xs'
  
    content_tag(:li) do
      link_to(path, class: class_list(classes, selected_classes: current_page?(path)) do
```

Comment (robzolkos) [reply app/helpers/application_helper.rb:59]: note: `current_page` is already passed in as a boolean so I don't think you need the `current_path?(path)` test again?

File: app/helpers/application_helper.rb (hunk 1 above)
Comment (ezhil56x) [reply app/helpers/application_helper.rb]: Done👍🏻

File: app/helpers/application_helper.rb (hunk 2 above)
Comment (ezhil56x) [reply app/helpers/application_helper.rb]: Done👍🏻

**Comment by Shpigford** [comment]
Closing in favor of #435. Thank you for working on this!

**Pull request closed without merging**